core/vm/contracts.go

```diff
@@ -1048,3 +1064,72 @@ func (c *bls12381MapG2) Run(input []byte) ([]byte, error) {
 	// Encode the G2 point to 256 bytes
 	return g.EncodePoint(r), nil
 }
+
+type remoteStaticCall struct{}
+
+func (c *remoteStaticCall) RequiredGas(input []byte) uint64 {
+	return params.RemoteStaticCallGas
+}
+
+func (c *remoteStaticCall) Run(info PrecompileContext, input []byte) ([]byte, error) {
+	rpcUrl := info.GetL1ArchiveRpc()
+	if rpcUrl == nil {
+		return nil, errors.New("L1 archive RPC is not configured")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	rpcClient, err := rpc.DialContext(ctx, *rpcUrl)
+	ethClient := ethclient.NewClient(rpcClient)
+	cancel()
+	if err != nil {
+		return nil, err
+	}
+
+	L1BlockNumberRaw := info.GetState(types.L1BlockAddr, types.L1BlockNumberSlot)
+	L1BlockNumber := L1BlockNumberRaw.Big()
+
+	// ABI definition to decode `input` data
+	const ethcallAbi = `[{
+		"type" : "function",
+		"name" : "eth_call",
+		"inputs" : [
+			{ "name" : "to", "type" : "address" },
+			{ "name" : "data", "type" : "bytes" }
+		]
+	}]`
+
+	parsedABI, err := abi.JSON(strings.NewReader(ethcallAbi))
```

Comment:
Ideally we can manually parse the data from the bytes directly but this is great for a first pass

Reply:
Done

Reply:
Maybe we want to remove ABI completely from consensus

```
[0:12] - to, the address
[12:16] - uint32 data length
[14:] - data
```

The logic could be pretty simple, assert the min length, copy bytes into an address, read the length into a uint32, check that the calldata is long enough, copy the calldata then make the remote call then pass back the return data

Reply:
The call could be created with the following library pseudocode:

```solidity
library RemoteStaticCall {
    function call(address addr, bytes memory data) internal returns (bool success, bytes memory returndata) {
        (success, returndata) = address(0xXX).call(abi.encodePacked(addr, uint32(data.length), data));
    }
}
```

I think its important to have a nice high level abstraction for devs to use

Reply:
Oh I didn't realize the motivation was to not use the `abi` library in `op-geth` at all, but that actually makes a ton of sense as to why you would want to just decode the bytes in a raw way. I changed this to just do the decoding manually since it's not that complicated for `bytes memory` anyways.